### PR TITLE
Fix: pinning cert check not work on M1 Mac

### PR DESCRIPTION
**Pull request by billypchan**
### Issues

On M1 Mac wire iOS app, network request always fails. 

### Causes

`SecTrustEvaluate` gives error on M1 Mac.
`SecTrustCopyPublicKey` does not return a key on M1 Mac.

### Solutions

Update the functions and remove `SecTrustEvaluate` check for the leaf certificate, which the result is error for missing intermediate node.
Refactor the code to Swift.

### notice
add Github action for testing with iOS 13 simulator to make sure `SecTrustCopyPublicKey ` still works

### CODE REVIEW

**Review by typfel — APPROVED** [review]

File: Source/Public/ServerCertificateTrust.swift [hunk 1]
```diff
@@ -34,8 +34,71 @@ class ServerCertificateTrust: NSObject, BackendTrustProvider {
...
+            return nil
+        }
+        
+        let key: SecKey?
+        
+        if #available(iOS 14.0, *) {
+            key = SecTrustCopyKey(trust)
+        } else {
+            var result: SecTrustResultType = SecTrustResultType.invalid
+            if SecTrustEvaluate(trust, &result) != noErr {
```
Comment (typfel) [inline Source/Public/ServerCertificateTrust.swift:78]: So this is actually required before running `SecTrustCopyPublicKey()` which explains why it was seemingly unnecessarily evaluating the certificate chain.

I guess it's not required `SecTrustCopyKey()` but it also zero documentation :-/

**Review by typfel — APPROVED** [review]

**Review by johnxnguyen — APPROVED** [review]

**Comment by billypchan** [comment]
@zenkins release major

**Comment by typfel** [comment]
@billypchan why did you release this as a major release? I thought there were no API changes.

**Comment by billypchan** [comment]
Since a public header is removed, i have to make sure that no downstream repo is affected

**Comment by typfel** [comment]
OK

**Pull request merged**